[buddy] CLI reference: add env var config and frontmatter

ptgott opened the pull request:
This change edits the CLI reference generator to fill in environment variables that are currently present only in the manually maintained CLI reference documentation, beginning with `tsh`. If the generator finds a YAML file named after the CLI tool it is generating a page for, it loads that file and adds the environment variables listed there to the complete list of environment variables for the CLI tool. This way, if a CLI reads environment variables directly from the OS, rather than from the `kingpin` library, we can still include these variables in the docs.

This change also adds the `sidebar_label` and `tags `frontmatter fields to each reference page.

aatuvai committed:
CLI reference generator: load default env variables from a YAML file

aatuvai committed:
- Add the 'sidebar_label' and 'tags' frontmatter fields
- Adjust the template and anyEnvVarsForCmd to prevent listing empty environment variable / flag lists

aadc-dev commented on lib/utils/usage_docs.go:
```diff
@@ -263,9 +266,91 @@ func updateAppUsageTemplate(r io.Reader, app *kingpin.Application) {
+func loadDefaultEnvVars(appName string) ([][4]string, error) {
+	pathname := filepath.Join("lib", "utils", "docenvdefaults", appName+".yaml")
+	data, err := os.ReadFile(pathname)
+	if err != nil || len(data) == 0 {
```
nit: it seems we are omitting errors from os.ReadFile on purpose, but we might benefit from adding some logs.
e.g.

```
if err != nil {
    if errors.Is(err, fs.ErrNotExist) {
        slog.Info("file does not exists", "path", filename)
    } else {
        slog.Info("unexpected read failure", "error", err
    }
    return 
}
```

followed a separate check on `len(data) == 0` check

aadc-dev reviewed (approved):
Left some questions regarding error handling when trying to open default environment variables.
If it's not ok to omit an `appName`  that yields no file, we should explicitly handle that.

ptgott committed:
Clean up the CLI doc generator

Remove unnecessary intermediate values: pass the unmarshaled YAML data
structure directly from `loadDefaultEnvVars` to `UpdateAppUsageTemplate`
without converting it from a `[][4]string`.

ptgott committed:
Improve CLI doc generator error handling

Instead of silently exiting with no error if it is not possible to read
the CLI doc generator config file, print an error message.

If there is no CLI generator config file, skip manual environment
variable additions and print a message. (Not using structured logging
since this will only ever be run manually and in CI.)
